examples/async.rs

```diff
@@ -1,16 +1,25 @@
 extern crate futures;
 extern crate reqwest;
 extern crate tokio_core;
+extern crate tokio_io;
 
+use std::io;
 use futures::Future;
 
 fn main() {
     let mut core = tokio_core::reactor::Core::new().unwrap();
     let client = reqwest::unstable::async::Client::new(&core.handle()).unwrap();
 
-    let work = client.get("https://hyper.rs").unwrap().send().map(|res| {
-        println!("{}", res.status());
-    });
+    let work = client.get("http://localhost:8080/range/4096?duration=5&chunk_size=300")
```

Comment:
Thanks @echochamber! That's a great idea. Debugging has been tricky, since there are a few permutations of this code that behave differently. I need to get some proper tests in there.